crates/autopilot/src/database/auction.rs

```diff
@@ -90,8 +95,20 @@ impl Postgres {
             .await?;
         let latest_settlement_block =
             database::orders::latest_settlement_block(&mut ex).await? as u64;
+        let quotes = {
+            let mut quotes = HashMap::new();
+            for order in &orders {
+                if let Some(quote) =
+                    database::orders::read_quote(&mut ex, &ByteArray(order.metadata.uid.0)).await?
```

Comment:
I think we already mentioned this somewhere. Anyway, implementing it in a single query right now is double work, since we also want to refactor solvable orders so that it:
1. fetch open orders but just a subset of data (compared to current OPEN ORDERS) needed for autopilot.
2. fetch other needed data in the same single query (quotes, whatever else).

So my plan was to merge it as is, and observe the performance and if ok, leave it that way until refactoring of solvable orders.

Reply:
Isn't it trivial though to expose a `read_quotes` method that fetches everything in a single query so we avoid up to 2k roundtrips per order? You mentioned sharing the connection somewhere before as a performance improvement (which made a lot of sense to me), but if we still do 2k roundtrips reusing the connection isn't going to help much, is it?

With observing performance the issue is that we will only see problems in prod, so we may loose valuable time to shipping.

Reply:
If we want a single query will need to build the query manually. To unblock https://github.com/cowprotocol/services/pull/2118 I am starting to work on this in a separate PR.

Reply:
https://github.com/cowprotocol/services/issues/2257